### PR TITLE
Tweaked cpp stress client flags

### PR DESCRIPTION
**Pull request by ncteisen**
Modified cpp stress client flags to conform with the recent changes made
to the Java and Python client. Stress client now supports tls and custom
ca file, but no longer contains both the test_cases and test_case flag. I did
this by using the create_test_channel file instead of client_helper (which uses
create_test_channel itself).

Also re-ran tools/buildgen/generate_project.sh to updated the Makefile.

### CODE REVIEW

File: test/cpp/interop/stress_test.cc [hunk 1]
```diff
@@ -45,11 +45,11 @@
 
 #include "src/proto/grpc/testing/metrics.grpc.pb.h"
 #include "src/proto/grpc/testing/metrics.pb.h"
-#include "test/cpp/interop/client_helper.h"
 #include "test/cpp/interop/interop_client.h"
 #include "test/cpp/interop/stress_interop_client.h"
 #include "test/cpp/util/metrics_server.h"
 #include "test/cpp/util/test_config.h"
+#include "test/cpp/util/create_test_channel.h"
```
Comment (ericgribkoff) [inline test/cpp/interop/stress_test.cc]: This should be moved up above metrics_server.h (alphabetical ordering)

File: test/cpp/interop/stress_test.cc [hunk 2]
```diff
@@ -340,12 +293,7 @@ int main(int argc, char** argv) {
          channel_idx++) {
       gpr_log(GPR_INFO, "Starting test with %s channel_idx=%d..", it->c_str(),
               channel_idx);
-      std::shared_ptr<grpc::Channel> channel;
-      if (FLAGS_use_tls) {
-        channel = grpc::testing::CreateChannelForTestCase(FLAGS_test_case);
-      } else {
-        channel = grpc::CreateChannel(*it, grpc::InsecureChannelCredentials());
-      }
+      std::shared_ptr<grpc::Channel> channel = grpc::CreateTestChannel(*it, FLAGS_server_host_override, FLAGS_use_tls, !FLAGS_use_test_ca);
```
Comment (ericgribkoff) [inline test/cpp/interop/stress_test.cc]: I think this line is too long

**Comment by ericgribkoff** [comment]
retest this please

**Comment by ncteisen** [comment]
Addressed all comments

**Comment by ncteisen** [comment]
retest this please

**Comment by ncteisen** [comment]
Merged in the fixit changes. Once that is declared over, this should be good to go

**Comment by ncteisen** [comment]
This cannot be merged until the build moratorium is over

**Comment by ncteisen** [comment]
retest this please

**Comment by ncteisen** [comment]
Failed tests:
aggregate_tests.run_tests_sanity_linux_opt
aggregate_tests.run_tests_sanity_linux_dbg
sanity_linux_dbg.tools/buildgen/generate_projects.sh
sanity_linux_opt.tools/buildgen/generate_projects.sh

These failures will be fixed with #9339

**Comment by ncteisen** [comment]
retest this please

**Comment by ncteisen** [comment]
Two failures:
c_windows_opt.vsprojects/Release/h2_http_proxy_test.exe connectivity
c_macos_dbg.bins/dbg/h2_compress_nosec_test resource_quota_server

Both are known flakes. See:
#9088
#9398 

Ready to be merged once it gets approved.

**Pull request merged**